src/app/lib/access/logger/logger_repository.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// Contract repository to handle logging configuration.
+abstract interface class LoggerRepository {
+  factory LoggerRepository() = _LoggerRepository;
+
+  /// Loads the current logging configuration.
+  Future<LoggingConfiguration> getLoggingConfiguration();
+
+  /// Sets whether console logging should be enabled on next app launch.
+  Future<bool> setIsConsoleLoggingEnabled(bool isConsoleLoggingEnabled);
```

Comment:
Why are the return types for the set methods `Future<bool>`?

Reply:
Essentially, because `sharedpreference.setbool` returns a `bool` to specify whether the value has been persisted. I figured that I do the same so we can act when it fails

Reply:
To be fair, I could document it

Reply:
When does that fail? What should we do?

Reply:
To be honest, I'm not quite sure. I looked at the github and it's not that clear either on their documentation.

Related github issue 
https://github.com/flutter/flutter/issues/146070
https://github.com/flutter/flutter/issues/124800

Reply:
For what we should. I would assume at least warn the user that it failed. Which looking at my code I forgot to do

Reply:
We're now showing a native prompt when it returns false

Reply:
Is that based on the Shared Preference implementation? In our own interface, do we want to expose those return values? Do we want all implementations to have those return values?

Reply:
Yeah, it's pretty much based on the implementation. About whether we should still return that `bool`, I assumed it would be a good idea for debugging purposes to have that information, however, we could also just throw an exception instead, but I wasn't sure if it's the recommended approach to handle errors on flutter/dart

Reply:
As long as we manage it, can't be much worse than .NET, even though I don't really know.

Reply:
In which conditions is this returning false? I really wonder if we want this in our contract. Having a return value implies having code that checks it and deals with the error case explicitly. I'm not sure if that's what we want.

Reply:
yeah if it's not even clear what that bool means, I think JP is right there and it just doesn't make sense to incorporate it into our code.

Reply:
I removed the `bool`. Instead I'm throwing an exception when it returns false. I've also applied that fix for the environment repository